atomicapp/providers/openshift.py

```diff
@@ -17,473 +17,188 @@
-                          "openshift.io/deployment-config.name=%s" % name}
-                url = self._get_url(namespace,
-                                    "replicationcontroller",
-                                    params=params)
-                (status_code, return_data) = \
-                    Utils.make_rest_request("get", url, verify=self.oc._requests_tls_verify())
-                if status_code != 200:
-                    raise ProviderFailedException("Cannot get Replication"
-                                                  "Controllers for Deployment"
-                                                  "Config %s (status code %s)" %
-                                                  (name, status_code))
-                # kind of returned data is ReplicationControllerList
-                # https://docs.openshift.com/enterprise/3.1/rest_api/kubernetes_v1.html#v1-replicationcontrollerlist
-                # we need modify items to get valid ReplicationController
-                items = return_data["items"]
-                for item in items:
-                    item["kind"] = "ReplicationController"
-                    item["apiVersion"] = return_data["apiVersion"]
-                # add items to list of artifact to be deleted
-                delete_artifacts.extend(items)
-
-            url = self._get_url(namespace, kind, name)
-
-            # Scale down replication controller to 0 replicas before deleting.
-            # This should take care of all pods created by this replication
-            # controller and we can safely delete it.
-            if kind.lower() == "replicationcontroller":
-                if self.dryrun:
-                    logger.info("DRY-RUN: SCALE %s down to 0", url)
-                else:
-                    self.oc.scale(url, 0)
+        default_config_loc = os.path.join(
+            Utils.getRoot(), Utils.getUserHome().strip('/'), '.kube/config')
 
-            if self.dryrun:
-                logger.info("DRY-RUN: DELETE %s", url)
-            else:
-                self.oc.delete(url)
+        if self.config_file:
+            logger.debug("Provider configuration provided")
+            self.api = Client(KubeConfig.from_file(self.config_file), "openshift")
+        elif self._check_required_params():
+            logger.debug("Generating .kube/config from given parameters")
+            self.api = Client(self._from_required_params(), "openshift")
+        elif os.path.isfile(default_config_loc):
+            logger.debug(".kube/config exists, using default configuration file")
+            self.api = Client(KubeConfig.from_file(default_config_loc), "openshift")
+        else:
+            self.config["provider-api"] = OC_DEFAULT_API
+            self.api = Client(self._from_required_params(), "openshift")
+
+        # Check if the namespace that the app is being deployed to is available
+        # DISABLED at the moment... Opened up issue against OpenShift..
```

Comment:
really would like to have the issue opened and referenced in this PR before we merge.